src/app/datepicker/datepicker.directive.view.spec.ts

```diff
@@ -0,0 +1,473 @@
+              [options]="options"
+              [(ngModel)]="date">
+          </mz-datepicker-container>
+        `, {
+          date: null,
+          options: { format: 'yyyy-mm-dd' },
+        }).then((fixture) => {
+          component = fixture.componentInstance;
+          nativeElement = fixture.nativeElement;
+          fixture.detectChanges();
+          tick();
+
+          datepicker().set('select', '2017-02-03');
+          tick();
+
+          expect(component.date).toBe('2017-02-03');
+          expect(input().value).toBe('2017-02-03');
+        });
+      }));
+
+      it('when formatSubmit is provided', fakeAsync(() => {
+
+        buildComponent<any>(`
+          <mz-datepicker-container>
+            <input mz-datepicker
+              id="datepicker"
+              [options]="options"
+              [(ngModel)]="date">
+          </mz-datepicker-container>
+        `, {
+          date: null,
+          options: { formatSubmit: 'yyyy-mm-dd' },
+        }).then((fixture) => {
+          component = fixture.componentInstance;
+          nativeElement = fixture.nativeElement;
+          fixture.detectChanges();
+          tick();
+
+          datepicker().set('select', '2017-02-03');
+          tick();
+
+          expect(component.date).toBe('2017-02-03');
+          expect(input().value).toBe('2017-02-03');
+        });
+      }));
+
+      it('when format and formatSubmit are provided', fakeAsync(() => {
+
+        buildComponent<any>(`
+          <mz-datepicker-container>
+            <input mz-datepicker
+              id="datepicker"
+              [options]="options"
+              [(ngModel)]="date">
+          </mz-datepicker-container>
+        `, {
+          date: null,
+          options: {
+            format: 'd mmmm, yyyy',
+            formatSubmit: 'yyyy-mm-dd',
+          },
+        }).then((fixture) => {
+          component = fixture.componentInstance;
+          nativeElement = fixture.nativeElement;
+          fixture.detectChanges();
+          tick();
+
+          datepicker().set('select', '2017-02-03', { format: component.options.formatSubmit });
+          tick();
+
+          expect(component.date).toBe('2017-02-03');
+          expect(input().value).toBe('3 February, 2017');
+        });
+      }));
+    });
+
+    describe('should set selected date correctly', () => {
+
+      it('when ngControl is set with a value', fakeAsync(() => {
+
+        buildComponent<any>(`
+          <mz-datepicker-container>
+            <input mz-datepicker
+              id="datepicker"
+              [options]="options"
+              [(ngModel)]="date">
+          </mz-datepicker-container>
+        `, {
+          date: null,
+          options: {
+            format: 'd mmmm, yyyy',
+            formatSubmit: 'yyyy-mm-dd',
+          },
+        }).then((fixture) => {
+          component = fixture.componentInstance;
+          nativeElement = fixture.nativeElement;
+          fixture.detectChanges();
```

Comment:
You can use `autoDetectChanges()` function for your tests.

Reply:
I tried it but tests that uses `tick()` to force `setTimeout` executions do not work if I change it for `autoDetectChanges()` ... all the other tests only use one `detectChanges()` so it wouldn't change anything